src/player.cpp

```diff
@@ -7775,27 +7779,35 @@ bool player::can_unwield( const item& it, bool alert ) const
 
 bool player::wield( item& target )
 {
-    if( !can_unwield( weapon ) || !can_wield( target ) ) {
-        return false;
-    }
-
     if( target.is_null() ) {
-        return dispose_item( item_location( *this, &weapon ),
-                             string_format( _( "Stop wielding %s?" ), weapon.tname().c_str() ) );
+        debugmsg( "Tried to wield an empty item." );
```

Comment:
Wielding a null item may be used in more places. If you want to change the meaning of `wield` to exclude this case, make sure you handle all invocations of `wield`. Be sure to check NPC functions.